src/Microsoft.ML.Data/Transforms/TermTransform.cs

```diff
@@ -175,32 +192,180 @@ private static VersionInfo GetTermManagerVersionInfo()
-                    Interlocked.CompareExchange(ref _codecFactory, new CodecFactory(Host, _codecFactoryPool), null);
+                    _textMetadata[iinfo] = _columns[iinfo].TextKeyValues;
                 }
-                Host.Assert(_codecFactory != null);
-                return _codecFactory;
+                ch.Assert(_unboundMaps.Length == _columns.Length);
+                ch.Done();
             }
         }
 
-        public override bool CanSavePfa => true;
-        public override bool CanSaveOnnx => true;
+        // Factory method for SignatureDataTransform.
+        public static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(args, nameof(args));
+            env.CheckValue(input, nameof(input));
+
+            env.CheckValue(args.Column, nameof(args.Column));
+            var cols = new ColumnInfo[args.Column.Length];
+            using (var ch = env.Start("ValidateArgs"))
+            {
+                if ((args.Term != null || !string.IsNullOrEmpty(args.Terms)) &&
+                  (!string.IsNullOrWhiteSpace(args.DataFile) || args.Loader != null ||
+                      !string.IsNullOrWhiteSpace(args.TermsColumn)))
+                {
+                    ch.Warning("Explicit term list specified. Data file arguments will be ignored");
+                }
+                if (!Enum.IsDefined(typeof(SortOrder), args.Sort))
+                    throw ch.ExceptUserArg(nameof(args.Sort), "Undefined sorting criteria '{0}' detected", args.Sort);
+
+                for (int i = 0; i < cols.Length; i++)
+                {
+                    var item = args.Column[i];
+                    var sortOrder = item.Sort ?? args.Sort;
+                    if (!Enum.IsDefined(typeof(SortOrder), sortOrder))
+                        throw env.ExceptUserArg(nameof(args.Sort), "Undefined sorting criteria '{0}' detected for column '{1}'", sortOrder, item.Name);
+
+                    cols[i] = new ColumnInfo(item.Source,
+                        item.Name,
+                        item.MaxNumTerms ?? args.MaxNumTerms,
+                        sortOrder,
+                        item.Term,
+                        item.Terms,
+                        item.TextKeyValues ?? args.TextKeyValues);
+                };
+            }
+            return new TermTransform(env, input, cols, args.DataFile, args.TermsColumn, args.Loader).MakeDataTransform(input);
+        }
+
+        public static TermTransform Create(IHostEnvironment env, ModelLoadContext ctx)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register(RegistrationName);
+
+            host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel(GetVersionInfo());
+
+            return new TermTransform(host, ctx);
+        }
+
+        private TermTransform(IHost host, ModelLoadContext ctx)
```

Comment:
>TermTransform [](start = 16, length = 13)

_columns are not initialized here.

I think they shouldn't be part of the transformer altogether